sklearn/decomposition/tests/test_truncated_svd.py

```diff
@@ -191,3 +192,24 @@ def test_truncated_svd_eq_pca(X_sparse):
     assert_allclose(Xt_svd, Xt_pca, rtol=1e-9)
     assert_allclose(pca.mean_, 0, atol=1e-9)
     assert_allclose(svd.components_, pca.components_)
+
+
+def test_fit_transform(X_sparse):
+    # when: 1, algorithm="randomized", 2, algorithm="arpack" with tol > 0,
+    # fit_transform(X) should equal to fit(X).transform(X)
```

Comment:
I think, with your changes, it should always give the same result and we should test that.
In that case, it could make sense to parametrize this test with `@pytest.mark.parametrize`.